File: src/BaseHmiTypes/Projects/HmiProjectInfo.cs
namespace BaseHmiTypes.Projects;

public class HmiProjectInfo
{
    public HmiProjectSoftwareType HmiProjectSoftwareType { get; set; }

    public string? ProjectTitle { get; set; }

    public string? DeviceFamilyString { get; set; }

    public string? ProjectName { get; set; }

    public string? ProjectVersionString { get; set; }

    public string? Author { get; set; }

    public string? EngineeringSoftwareVersionString { get; set; }

    public string? OriginalDateString { get; set; }

    public string? LastModifiedDateString { get; set; }
}

File: typescript/base-hmi-types/src/projects/HmiProjectInfo.ts
import { HmiProjectSoftwareType } from "./HmiProjectSoftwareType.js";

export class HmiProjectInfo {
  hmiProjectSoftwareType: HmiProjectSoftwareType = HmiProjectSoftwareType.Unknown;
  projectTitle?: string;
  deviceFamilyString?: string;
  projectName?: string;
  projectVersionString?: string;
  author?: string;
  engineeringSoftwareVersionString?: string;
  originalDateString?: string;
  lastModifiedDateString?: string;
}
